R/jqueryui.R

```diff
@@ -76,16 +76,25 @@ absolutePanel <- function(...,
 
   style <- paste(paste(names(cssProps), cssProps, sep = ':', collapse = ';'), ';', sep='')
   divTag <- tags$div(style=style, ...)
-  if (isTRUE(draggable)) {
-    divTag <- tagAppendAttributes(divTag, class='draggable')
-    return(tagList(
-      divTag,
-      jqueryuiDependency(),
-      tags$script('$(".draggable").draggable();')
-    ))
-  } else {
+
+  if (identical(draggable, FALSE)) {
     return(divTag)
   }
+
+  # CPS (2023-10-31): it'd be nice if draggable could be a list of options for
+  # the draggable() method, but to do that properly (i.e., support JS functions,
+  # per element options, etc) would be non-trivial. For now, we'll just add a
+  # smart cancel option default to prevent dragging on input/output elements.
+  # This is particularly important for widgets (e.g. plotly, leaflet) and
+  # selectize.js https://github.com/rstudio/shiny/issues/3752
+  # https://github.com/rstudio/shiny/issues/3933
```

Comment:
I think the intent is clear enough from the code, once you know what the `cancel` option does, so I rephrased this comment.

```suggestion
  # Prevent dragging from starting on shiny inputs or html widgets
  # Cf. https://api.jqueryui.com/draggable/#option-cancel
  #
  # CPS (2023-10-31): We should allow users to pass a list of options via the
  # `draggable` argument, but proper support (i.e. JS functions, per-element
  # options, etc) would be non-trivial.
```

I wonder if we're more likely to see the future dev idea if it's here in a comment or in an issue? I think discoverability is low on both but I lean towards issues since there easier to search. If you open an issue instead, this comment could be

```suggestion
  # Prevent dragging from starting on shiny inputs or html widgets
  # Cf. https://api.jqueryui.com/draggable/#option-cancel
```

Reply:
Another option is to just leave the note in this PR